test/e2e-tests.sh

```diff
@@ -63,9 +63,15 @@ go_test_e2e -timeout=10m \
 if [[ -n "${ISTIO_VERSION}" ]]; then
   go_test_e2e -timeout=10m \
     ./test/e2e/istio \
-    "--resolvabledomain=$(use_resolvable_domain)" "$(use_https)" || failed=1
+    "--resolvabledomain=$(use_resolvable_domain)" || failed=1
 fi
 
+# Serial tests for https
+go_test_e2e -timeout=10m \
+  -parallel 1
+  ./test/e2e \
+  "--resolvabledomain=$(use_resolvable_domain)" "$(use_https)" || failed=1
```

Comment:
Shouldn't this `go_test_e2e` call be inside an `if` statement on `$(use_https)`?
My understanding is that you want to run **only** the regular (non HTTPS) tests in the normal case (what we do today) and run the **only** the HTTPS tests when `use_https` is used? The only difference being the parallelism. In this case can't we do exactly the same thing that is done for `mesh` vs. `no_mesh`?

Reply:
That's right. `use_https` doesn't seem necessary anymore since it's only used in one place now. Having the call inside an if statement is better.

Reply:
The issue is still there, isn't it?
Now, HTTP E2E tests always run and HTTPS E2E tests run if HTTPS is enabled.
If w want mutual exclusion between HTTP and HTTPS, we should do something like
```suggestion
(( HTTPS )) && parallelism="-parallel 1"
(( HTTPS )) && use_https="--https"

go_test_e2e -timeout=30m \
  ./test/conformance/... \
  ./test/e2e \
  ${parallelism} \
  "--resolvabledomain=$(use_resolvable_domain)" "$(use_https)" "$(ingress_class)" || failed=1
```

A single call to `go_test_e2e -timeout=30m ./test/conformance/...`, parametrized depending on `HTTPS`.